### ISSUE TITLE
new version of six (1.11.0) breaks pgpy with python2.7

### ISSUE THREAD

**Issue by drebs** [SecurityInnovation/PGPy#217]
six [just released version 1.11.0](https://github.com/benjaminp/six/releases/tag/1.11.0), which  seems to break pgpy when using python2.7:

```sh
$ virtualenv -p python2.7 /tmp/venv
Running virtualenv with interpreter /usr/bin/python2.7
New python executable in /tmp/venv/bin/python2.7
Also creating executable in /tmp/venv/bin/python
Installing setuptools, pip, wheel...done.
$ source /tmp/venv/bin/activate
$ python --version
Python 2.7.13
$ pip install pgpy
Collecting pgpy
Collecting pyasn1 (from pgpy)
  Using cached pyasn1-0.3.5-py2.py3-none-any.whl
Collecting six>=1.9.0 (from pgpy)
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting singledispatch (from pgpy)
  Using cached singledispatch-3.4.0.3-py2.py3-none-any.whl
Collecting enum34 (from pgpy)
  Using cached enum34-1.1.6-py2-none-any.whl
Collecting cryptography>=1.1 (from pgpy)
  Using cached cryptography-2.0.3-cp27-cp27mu-manylinux1_x86_64.whl
Collecting ipaddress (from cryptography>=1.1->pgpy)
  Using cached ipaddress-1.0.18-py2-none-any.whl
Collecting idna>=2.1 (from cryptography>=1.1->pgpy)
  Using cached idna-2.6-py2.py3-none-any.whl
Collecting asn1crypto>=0.21.0 (from cryptography>=1.1->pgpy)
  Using cached asn1crypto-0.22.0-py2.py3-none-any.whl
Collecting cffi>=1.7 (from cryptography>=1.1->pgpy)
  Using cached cffi-1.11.0-cp27-cp27mu-manylinux1_x86_64.whl
Collecting pycparser (from cffi>=1.7->cryptography>=1.1->pgpy)
Installing collected packages: pyasn1, six, singledispatch, enum34, ipaddress, idna, asn1crypto, pycparser, cffi, cryptography, pgpy
Successfully installed asn1crypto-0.22.0 cffi-1.11.0 cryptography-2.0.3 enum34-1.1.6 idna-2.6 ipaddress-1.0.18 pgpy-0.4.3 pyasn1-0.3.5 pycparser-2.18 singledispatch-3.4.0.3 six-1.11.0
$ python -c "import pgpy"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/tmp/venv/local/lib/python2.7/site-packages/pgpy/__init__.py", line 5, in <module>
    from .pgp import PGPKey
  File "/tmp/venv/local/lib/python2.7/site-packages/pgpy/pgp.py", line 24, in <module>
    from .constants import CompressionAlgorithm
  File "/tmp/venv/local/lib/python2.7/site-packages/pgpy/constants.py", line 22, in <module>
    from .types import FlagEnum
  File "/tmp/venv/local/lib/python2.7/site-packages/pgpy/types.py", line 260, in <module>
    class PGPObject(six.with_metaclass(abc.ABCMeta, object)):
  File "/tmp/venv/lib/python2.7/abc.py", line 87, in __new__
    cls = super(ABCMeta, mcls).__new__(mcls, name, bases, namespace)
TypeError: Error when calling the metaclass bases
    metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

If I force six version 1.10.0 it works:

```sh
$ pip install six==1.10.0
Collecting six==1.10.0
  Using cached six-1.10.0-py2.py3-none-any.whl
Installing collected packages: six
  Found existing installation: six 1.11.0
    Uninstalling six-1.11.0:
      Successfully uninstalled six-1.11.0
Successfully installed six-1.10.0
$ python -c "import pgpy"
```

**Comment by jhford** [comment]
Hey, thanks for fixing this!  I was wondering what the time frame is for a new release which would have this patch included?

**Comment by Commod0re** [comment]
I was hoping six would have had a patch released by now to fix their forward compatibility (it sounded like they were planning on it), but since that has not happened, I will do another patch release that will include this fix very soon

**Comment by edmorley** [comment]
> I was hoping six would have had a patch released by now to fix their forward compatibility (it sounded like they were planning on it), but since that has not happened, I will do another patch release that will include this fix very soon

I've pinged benjaminp/six#210 again asking for a revert, however it's worth noting the usage here was broken, so the fix to PGPy was needed either way.

**Comment by Commod0re** [comment]
It actually worked prior to six 1.11.0, so it clearly was not "broken" prior to 1.11.0, even if it was not the intended usage. The "correct" code did not work as it was supposed to when I originally wrote that stuff, which is why it was that way. The supposedly "broken" code had no adverse side effects prior to six 1.11.0, so I'm not even sure what value the changes that caused it to become broken actually have.

**Comment by edmorley** [comment]
Hi! I don't suppose it would be possible to cut a new point release including the fix for this at some point soon? Many thanks :-)

**Comment by jhford** [comment]
heh, I just came here to check the same thing!

**Comment by Commod0re** [comment]
Working on it!

**Comment by edmorley** [comment]
If there's anything we can do to help, let us know :-)

**Comment by edmorley** [comment]
Perhaps to simplify the release, just these commits (and not the newer ones) could be released, reducing the amount of testing/prep required?
https://github.com/SecurityInnovation/PGPy/compare/0.4.3...28920bc

**Comment by NCommander** [comment]
Is there any chance we can get a fix released for this?

**Comment by sayoojsamuel** [comment]
This issue is popping back up now. I have `six==1.12.0` now, and the error message is 
```python
TypeError: Error when calling the metaclass bases
    metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```
>Please note that reverting the **six** version to `1.10.0` is causing other apps to crash
Any plans to reopen this issue?

**Comment by Commod0re** [comment]
this is fixed and will be available shortly in PGPy v0.5.0
